src/libs/conduit/conduit_data_type.cpp

```diff
@@ -552,6 +552,15 @@ DataType::spanned_bytes() const
     return m_offset + m_stride * (m_num_ele -1) + m_ele_bytes;
 }
 
+//---------------------------------------------------------------------------//
+bool
+DataType::convertible(const DataType& dtype) const
+{
+    return ( (group() == dtype.group()) &&
+             (m_ele_bytes >= dtype.m_ele_bytes) &&
+             (m_num_ele >= dtype.m_num_ele) );
+}
```

Comment:
I believe any numeric type can be converted to any other numeric.  Seems like a property of just the type id, not the full DataType.

Seems odd we check the bytes and the number of elements?

Does this try to capture if data will be lost? Again the semantics are simply that of a c cast, creating our own semantics gets confusing.

Reply:
> I believe any numeric type can be converted to any other numeric. Seems like a property of just the type id, not the full DataType.

Yeah, it seems like "convert" really isn't the right verb here. Due to this ambiguity, I've removed this public interface and moved the functionality into the private `find_widest_type` function instead.

> Seems odd we check the bytes and the number of elements?

In order for type A to be "upscalable" to type B (or whatever the right verb is), these types must target the same form of data (e.g. floats, unsigned integers, signed integers, etc.) and type A's data needs to fit into type B. In order to fit, type B needs to be larger in both per-element bytes (or else there's the potential for per-element truncation) **and** total number of elements (or else there's the possibility of array value truncation).

> Does this try to capture if data will be lost? Again the semantics are simply that of a c cast, creating our own semantics gets confusing.

As I envision it, "upscaling" a type comes with no possibility of data loss because the upscaled type will always have at least as many bytes as the type being upscaled.